controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -370,7 +369,7 @@ func (r *DataScienceClusterReconciler) updateComponents(ctx context.Context, ori
 	return saved, err
 }
 
-func (r *DataScienceClusterReconciler) watchDataScienceClusterResources(a client.Object) (requests []reconcile.Request) {
+func (r *DataScienceClusterReconciler) watchDataScienceClusterResources(a client.Object) (requests []reconcile.Request) { //nolint:nonamedreturns // We want to capture all errors.
```

Comment:
I'm confused here. We do not have a variable `requests` in use in the code below. We always newly created slice - therefore there's no need for named return at all.

```suggestion
func (r *DataScienceClusterReconciler) watchDataScienceClusterResources(a client.Object) []reconcile.Request { ```

Should work.